synapse/app/_base.py

```diff
@@ -528,3 +530,25 @@ def sdnotify(state):
         # this is a bit surprising, since we don't expect to have a NOTIFY_SOCKET
         # unless systemd is expecting us to notify it.
         logger.warning("Unable to send notification to systemd: %s", e)
+
+
+def max_request_body_size(config: HomeServerConfig) -> int:
```

Comment:
I'm not terribly happy with this, but couldn't think of anywhere better to do it.